src/Util/ErrorFormatGuesser.php

```diff
@@ -37,14 +37,35 @@ private function __construct()
     public static function guessErrorFormat(Request $request, array $errorFormats): array
     {
         $requestFormat = $request->getRequestFormat('');
-        if ('' !== $requestFormat && isset($errorFormats[$requestFormat])) {
+
+        if ('' !== $requestFormat && array_key_exists($requestFormat, $errorFormats)) {
             return ['key' => $requestFormat, 'value' => $errorFormats[$requestFormat]];
         }
 
-        foreach ($errorFormats as $key => $value) {
-            return ['key' => $key, 'value' => $value];
+        $mimeType = self::getMimeType($request);
+
+        foreach ($errorFormats as $format => $mimeTypes) {
+            if (in_array($mimeType, $mimeTypes, true)) {
+                return ['key' => $format, 'value' => $mimeTypes];
+            }
+        }
+
+        return ['key' => key($errorFormats), 'value' => reset($errorFormats)];
```

Comment:
Because we already loop over formats, you can just store the first format in a var, line 48:

```php
$requestMimeTypes = $request->getMimeTypes($request->getRequestFormat());
$defaultFormat = [];
foreach ($errorFormats as $format => $errorMimeTypes) {
    if ([] === $defaultFormat) {
        $defaultFormat = ['key' => $format, 'value' => $mimeTypes];
    }

    if (array_intersect($requestMimeTypes, $errorMimeTypes) {
        return ['key' => $format, 'value' => $mimeTypes];
    }
}

return $defaultFormat;
```

It's faster and safer in case no default format is defined.